### PR TITLE
Rem 221

### PR DESCRIPTION
**Pull request by Alinadimmokdad**

### CODE REVIEW

**Comment by NoraBitarSM** [comment]
functionality tested

File: src/components/Shared/GridToolbar.js [hunk 1]
```diff
@@ -111,3 +123,27 @@ const GridToolbar = ({
 }
 
 export default GridToolbar
+
+{
+  /* <Tooltip title={button.key} key={index}>
+<Button
+  onClick={handleClick}
+  variant='contained'
```
Comment (Ghinwa-Arab) [inline src/components/Shared/GridToolbar.js:131]: Remove (if it's not needed anymore)

File: src/providers/AuthContext.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ const AuthProvider = ({ children }) => {
     const fetchData = async () => {
       const matchHostname = window.location.hostname.match(/^(.+)\.softmachine\.co$/)
 
-      const accountName = matchHostname ? matchHostname[1] : 'byc-deploy'
+      const accountName = matchHostname ? matchHostname[1] : 'neg2x'
```
Comment (Ghinwa-Arab) [inline src/providers/AuthContext.js:70]: please return it back byc-deploy

File: src/pages/plant-groups/index.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    }
+  ]
+
+  const add = () => {
+    setWindowOpen(true)
+  }
+
+  const edit = obj => {
+    setSelectedRecordId(obj.recordId)
+    setWindowOpen(true)
```
Comment (Ghinwa-Arab) [inline src/pages/plant-groups/index.js:87]: It's better to delete setWindowOpen state and use (useWindow) to open the form

File: src/pages/plant-groups/index.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    setWindowOpen(true)
+  }
+
+  const edit = obj => {
+    setSelectedRecordId(obj.recordId)
```
Comment (Ghinwa-Arab) [inline src/pages/plant-groups/index.js:86]: no need for setSelectedRecordId anymore

File: src/pages/plant-groups/Windows/PlantGroupsWindow.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+// ** Custom Imports
+import Window from 'src/components/Shared/Window'
```
Comment (Ghinwa-Arab) [inline src/pages/plant-groups/Windows/PlantGroupsWindow.js:2]: no need for this file anymore

File: src/pages/plant-groups/index.js [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
+          maxAccess={access}
+        />
+      </Box>
+      {windowOpen && (
+        <PlantWindow
```
Comment (Ghinwa-Arab) [inline src/pages/plant-groups/index.js:131]: No need for PlantComponentWindow delete it and open PlantGroupsForm directly by using useWindow

Comment (Alinadimmokdad) [reply src/pages/plant-groups/index.js:131]: done

File: src/pages/plant-groups/index.js (hunk 3 above)
Comment (Alinadimmokdad) [reply src/pages/plant-groups/index.js:87]: done

File: src/components/Shared/Tree.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+  '@media print': {
+    height: 'auto',
+    maxHeight: '100%',
+    width: '100%',
+    maxWidth: '100%',
+    overflowY: 'visible'
+  }
+}))
+console.log('Streeview', StyledTreeView)
```
Comment (omarZrayka23) [inline src/components/Shared/Tree.js:59]: clear console

File: src/pages/plant-groups/index.js [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
+          maxAccess={access}
+        />
+      </Box>
+
+      <ErrorWindow open={errorMessage} onClose={() => setErrorMessage(null)} message={errorMessage} />
```
Comment (omarZrayka23) [inline src/pages/plant-groups/index.js:138]: ba3dna 3am nsta3ml  ErrorWindow

File: src/pages/plant-groups/index.js [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+
+    {
+      field: 'parentName',
+      headerName: _labels.parent,
+      flex: 1
+    }
+  ]
+
+  const add = () => {
+    openForm('')
```
Comment (omarZrayka23) [inline src/pages/plant-groups/index.js:76]: openForm() i sgood without ''

File: src/pages/plant-groups/index.js [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+import ErrorWindow from 'src/components/Shared/ErrorWindow'
+import { useInvalidate, useResourceQuery } from 'src/hooks/resource'
+
+// ** Resources
+import { ResourceIds } from 'src/resources/ResourceIds'
+
+const Plant = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const [errorMessage, setErrorMessage] = useState(null)
```
Comment (omarZrayka23) [inline src/pages/plant-groups/index.js:27]: ?

File: src/pages/plant-groups/forms/PlantGroupsForm.js [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+            label={labels.parent}
+            valueField='recordId'
+            displayField={'name'}
+            columnsInDropDown={[
+              { key: 'reference', value: 'Reference' },
+              { key: 'name', value: 'Name' }
+            ]}
+            values={formik.values}
+            onChange={(event, newValue) => {
+              formik.setFieldValue('parentId', newValue?.recordId)
```
Comment (omarZrayka23) [inline src/pages/plant-groups/forms/PlantGroupsForm.js:129]: onClear is work?

Comment (Alinadimmokdad) [reply src/pages/plant-groups/forms/PlantGroupsForm.js:129]: yes , i need it

File: src/components/Shared/GridToolbar.js [hunk 12]
```diff
@@ -18,6 +18,7 @@ const GridToolbar = ({
   initialLoad,
   onAdd,
   openRPB,
+  onTree,
   disableRPB = false,
```
Comment (hadichahine) [inline src/components/Shared/GridToolbar.js:22]: Let's add change this into actions.
`[{
image: '/images/buttonsIcons/tree.png',
onClick(){}
}]`

File: src/components/Shared/GridToolbar.js [hunk 13]
```diff
@@ -135,6 +136,17 @@ const GridToolbar = ({
             />
           </Box>
         )}
+        {onTree && (
```
Comment (hadichahine) [inline src/components/Shared/GridToolbar.js:139]: Loop over actions.

File: src/components/Shared/FormShell.js [hunk 14]
```diff
@@ -2,6 +2,7 @@ import { DialogContent, Box } from '@mui/material'
 import { useState } from 'react'
 import WindowToolbar from './WindowToolbar'
 import TransactionLog from './TransactionLog'
+import Tree from './Tree'
```
Comment (hadichahine) [inline src/components/Shared/FormShell.js:5]: Remove

File: src/components/Shared/Buttons.js [hunk 15]
```diff
@@ -73,12 +73,24 @@ export const Buttons = [
     image: 'beneficiary.png',
     main: false
   },
+  {
```
Comment (hadichahine) [inline src/components/Shared/Buttons.js:76]: Why do we have those in 2 places?

File: src/components/Shared/Tree.js [hunk 16]
```diff
@@ -0,0 +1,153 @@
...
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        height: '100%'
+      }}
+    >
+      <div>
+        <div ref={componentRef}>
+          <StyledTreeView
+            height={`${expanded ? `calc(100vh - 180px)` : `${height}px`}`}
```
Comment (hadichahine) [inline src/components/Shared/Tree.js:123]: Remove

File: src/components/Shared/Tree.js [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+  '&.Mui-selected > .MuiTreeItem-content .MuiTreeItem-label': {
+    backgroundColor: 'transparent',
+    borderColor: 'transparent'
+  },
+  '&.Mui-focused > .MuiTreeItem-content .MuiTreeItem-label': {
+    backgroundColor: 'transparent'
+  }
+}))
+
+function Tree({ data, expanded, height }) {
```
Comment (hadichahine) [inline src/components/Shared/Tree.js:70]: Try to remove expanded and height.

File: src/components/Shared/Tree.js [hunk 18]
```diff
@@ -0,0 +1,153 @@
...
+  const componentRef = useRef()
+  const printComponentRef = useRef()
+
+  const handlePrint = useReactToPrint({
+    content: () => printComponentRef.current,
+    documentTitle: 'Tree_Structure',
+    removeAfterPrint: true
+  })
+
+  useEffect(() => {
```
Comment (hadichahine) [inline src/components/Shared/Tree.js:81]: Try to remove it, or replace it with useMemo.

File: src/components/Shared/WindowToolbar.js [hunk 19]
```diff
@@ -22,6 +22,7 @@ const WindowToolbar = ({
   onClickGL,
   onGenerateReport,
   disabledSubmit,
+  print,
```
Comment (hadichahine) [inline src/components/Shared/WindowToolbar.js:25]: What's this?

File: src/pages/plant-groups/index.js [hunk 20]
```diff
@@ -0,0 +1,140 @@
...
+  }
+
+  const edit = obj => {
+    openForm(obj?.recordId)
+  }
+
+  return (
+    <VertLayout>
+      <Fixed>
+        <GridToolbar onAdd={add} maxAccess={access} onTree={onTreeClick} />
```
Comment (hadichahine) [inline src/pages/plant-groups/index.js:119]: This should be turned into actions.

File: src/pages/plant-groups/forms/PlantGroupsForm.js [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { useForm } from 'src/hooks/form'
+
+export default function PlantGroupsForm({ labels, maxAccess, recordId }) {
+  const [isLoading, setIsLoading] = useState(false)
```
Comment (hadichahine) [inline src/pages/plant-groups/forms/PlantGroupsForm.js:17]: Okay, check if we need this.

File: src/pages/plant-groups/forms/PlantGroupsForm.js [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+          <CustomTextField
+            name='reference'
+            label={labels.reference}
+            value={formik.values.reference}
+            required
+            rows={2}
+            maxAccess={maxAccess}
+            onChange={formik.handleChange}
+            onClear={() => formik.setFieldValue('reference', '')}
+            error={formik.touched.reference && Boolean(formik.errors.reference)}
```
Comment (hadichahine) [inline src/pages/plant-groups/forms/PlantGroupsForm.js:98]: Make sure that error handling works.

File: src/pages/bp-master-data/forms/RelationForm.js
Comment (hadichahine) [inline src/pages/bp-master-data/forms/RelationForm.js:1]: Check if we need these

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/plant-groups/forms/PlantGroupsForm.js [hunk 23]
```diff
@@ -0,0 +1,132 @@
...
+import { useInvalidate } from 'src/hooks/resource'
+import { ResourceIds } from 'src/resources/ResourceIds'
+
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import { useForm } from 'src/hooks/form'
+
+export default function PlantGroupsForm({ labels, maxAccess, recordId }) {
+  const [editMode, setEditMode] = useState(!!recordId)
```
Comment (hadichahine) [inline src/pages/plant-groups/forms/PlantGroupsForm.js:17]: Can we turn it into const editMode =  !!recordId

File: src/components/Shared/Tree.js [hunk 24]
```diff
@@ -0,0 +1,141 @@
...
+  '&.Mui-selected > .MuiTreeItem-content .MuiTreeItem-label': {
+    backgroundColor: 'transparent',
+    borderColor: 'transparent'
+  },
+  '&.Mui-focused > .MuiTreeItem-content .MuiTreeItem-label': {
+    backgroundColor: 'transparent'
+  }
+}))
+
+function Tree({ data, expanded, height }) {
```
Comment (hadichahine) [inline src/components/Shared/Tree.js:65]: Remove height and expanded please.

File: src/components/Shared/Tree.js [hunk 25]
```diff
@@ -0,0 +1,141 @@
...
+
+  return (
+    <>
+      <DialogContent
+        sx={{
+          padding: 0
+        }}
+      >
+        <StyledTreeView
+          height={`${expanded ? `calc(100vh - 180px)` : `${height}px`}`}
```
Comment (hadichahine) [inline src/components/Shared/Tree.js:116]: I removed this. Why is it here?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/plant-groups/forms/PlantGroupsForm.js (hunk 22 above)
Comment (Alinadimmokdad) [reply src/pages/plant-groups/forms/PlantGroupsForm.js:94]: done

**Review by hadichahine — APPROVED** [review]

**Pull request merged**